src/components/ModelingMachineProvider.tsx

```diff
@@ -791,7 +793,13 @@ export const ModelingMachineProvider = ({
             }
             const { modifiedAst, pathToNode: pathToNewSketchNode } = sketched
 
-            await kclManager.executeAstMock(modifiedAst)
+            const didReParse = await kclManager.executeAstMock(modifiedAst)
+            if (err(didReParse)) {
+              // there was a problem, restore the original code
+              codeManager.code = originalCode
+              await kclManager.executeCode()
+              return reject(didReParse)
+            }
```

Comment:
This is the fix here, but I had to refactor executeAstMock to return an error if it failed to parse the code, it was just silently failing before.

Reply:
Not sure I understand the reason for `executeCode()`.